download-utils/src/lib.rs

```diff
@@ -268,6 +268,11 @@ pub fn download_snapshot_archive<'a, 'b>(
         maximum_incremental_snapshot_archives_to_retain,
     );
 
+    let snapshot_archives_remote_dir = snapshot_archives_dir
+        .to_path_buf()
+        .join(snapshot_utils::SNAPSHOT_ARCHIVE_DOWNLOAD_DIR);
```

Comment:
Since this logic is going to be repeated in a few places, how about making it into its own function in snapshot_utils? Something like:

```rust
pub fn build_snapshot_archives_remote_dir(snapshot_archives_dir: impl Into<PathBuf>) -> PathBuf {
    snapshot_archives_dir
        .into()
        .join(SNAPSHOT_ARCHIVE_DOWNLOAD_DIR)
}
```

And then it called be called with a bunch of different inputs:

```rust
        let _ = build_snapshot_archives_remote_dir("string literal");

        let x = "str in var";
        let _ = build_snapshot_archives_remote_dir(x);

        let x = &Path::new("path borrowed");
        let _ = build_snapshot_archives_remote_dir(x);

        let x = Path::new("path moved");
        let _ = build_snapshot_archives_remote_dir(x);

        let x = &PathBuf::from_str("pathbuf borrowed").unwrap();
        let _ = build_snapshot_archives_remote_dir(x);

        let x = PathBuf::from_str("pathbuf moved").unwrap();
        let _ = build_snapshot_archives_remote_dir(x);
```

Reply:
Good idea.